StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Controllers/PayWithLinkViewController.swift

```diff
@@ -134,177 +151,61 @@ final class PayWithLinkViewController: UINavigationController {
 
     private init(context: Context) {
         self.context = context
-        super.init(nibName: nil, bundle: nil)
-
-        // Show loader
-        setRootViewController(LoaderViewController(context: context), animated: false)
+        super.init()
     }
 
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        view.accessibilityIdentifier = "Stripe.Link.PayWithLinkViewController"
-        view.tintColor = .linkBrand
-
-        // Hide the default navigation bar.
-        setNavigationBarHidden(true, animated: false)
-
-        // Apply the preferred user interface style.
-        context.configuration.style.configure(self)
-
-        updateSupportedPaymentMethods()
-        updateUI()
-
-        // The internal delegate of the interactive pop gesture disables
-        // the gesture when the navigation bar is hidden. Use a custom delegate
-        // to restore the functionality.
-        interactivePopGestureRecognizer?.delegate = self
-    }
-
-    override func viewWillDisappear(_ animated: Bool) {
-        super.viewWillDisappear(animated)
-    }
-
-    override func pushViewController(_ viewController: UIViewController, animated: Bool) {
-        if let viewController = viewController as? BaseViewController {
-            viewController.coordinator = self
-            viewController.customNavigationBar.linkAccount = linkAccount
-            viewController.customNavigationBar.showBackButton = !viewControllers.isEmpty
-        }
-
-        super.pushViewController(viewController, animated: animated)
-    }
-
-    private func updateUI() {
-        guard let linkAccount = linkAccount else {
-            if !(rootViewController is SignUpViewController) {
-                setRootViewController(
-                    SignUpViewController(linkAccount: nil, context: context)
-                )
+    var aswas: ASWebAuthenticationSession?
+    /// Defaults to the app's key window
+    func present(over viewController: UIViewController? = nil) {
```

Comment:
just checking, this needs to be nullable?  Shouldn't we always have a presenting view controller?

Reply:
Yep, will fix, we don't use it without a VC anywhere.